lib/nerves/utils/http_client.ex

```diff
@@ -78,6 +78,10 @@ defmodule Nerves.Utils.HTTPClient do
     {:noreply, %{s | url: url, caller: from, opts: opts, progress?: progress?}}
   end
 
+  def handle_info({:http, {_ref, {:error, {:failed_connect, [{:to_address, _address}, _reason]}}}}, s) do
+    GenServer.reply(s.caller, {:error, :failed_connect})
```

Comment:
should there be an error log of some kind for this, so whoever is building is aware this has happened?

Reply:
there is.. - the error is being returned at the upper instance